pkgs/by-name/db/dbeaver-bin/package.nix

```diff
@@ -55,18 +55,32 @@ stdenvNoCC.mkDerivation (finalAttrs: {
   dontConfigure = true;
   dontBuild = true;
 
-  prePatch = ''
-    substituteInPlace ${lib.optionalString stdenvNoCC.hostPlatform.isDarwin "Contents/Eclipse/"}dbeaver.ini \
-      --replace-fail '-Xmx1024m' '-Xmx${override_xmx}'
-  '';
-
-  preInstall = ''
-    # most directories are for different architectures, only keep what we need
-    shopt -s extglob
-    pushd ${lib.optionalString stdenvNoCC.hostPlatform.isDarwin "Contents/Eclipse/"}plugins/com.sun.jna_*/com/sun/jna/
-    rm -r !(ptr|internal|linux-x86-64|linux-aarch64|darwin-x86-64|darwin-aarch64)/
-    popd
-  '';
+  prePatch =
+    ''
+      substituteInPlace ${lib.optionalString stdenvNoCC.hostPlatform.isDarwin "Contents/Eclipse/"}dbeaver.ini \
+        --replace-fail '-Xmx1024m' '-Xmx${override_xmx}'
+    ''
+    # remove the bundled JRE configuration on Darwin
+    # dont use substituteInPlace here because it would match "-vmargs"
+    + lib.optionalString stdenvNoCC.hostPlatform.isDarwin ''
+      sed -i -e '/^-vm$/{
+          N
+          d
+      }' Contents/Eclipse/dbeaver.ini
```

Comment:
Anyway we can write the sed invocation in one line?

Reply:
got it!